ctk/idp/coverage/NotTested.md

```diff
@@ -0,0 +1,59 @@
+| 2.7.3.1 - Element <Attribute> | Any other uses of the element by profiles or other specifications MUST define the <Attribute> semantics of specifying or omitting elements. <AttributeValue> | Not verifying custom profiles or their specifications.
+| 2.7.3.1.1 - Element <AttributeValue> | If a SAML attribute includes an empty value, such as the empty string, the corresponding <AttributeValue> element MUST be empty | Impossible to know if an Attribute had an empty string value from a high level perspective, i.e. "empty".
+| 2.7.3.1.1 - Element <AttributeValue> | If a SAML attribute includes a "null" value, the corresponding <AttributeValue> element MUST be empty and MUST contain the reserved xsi:nil XML attribute with a value of "true" or "1". | Impossible to verify if an Attribute had a null value
+| 2.7.3.2 - Element <EncryptedAttribute> | The encrypted content MUST contain an element that has a type of or derived from AttributeType. | It would take a lot of effort to verify the type of the decrypted element and provides little value. However, the extended element's SAML defined base values will be verified by the schema validator.
+| 2.7.3.2 - Element <EncryptedAttribute> | Encrypted identifiers are intended as a privacy protection mechanism when the plain-text value passes through an intermediary. As such, the ciphertext MUST be unique to any given encryption operation. For more on such issues, see \[XMLEnc] Section 6.3. | This is implicitly verified by following XML Encryption rules when decrypting.
+| 2.7.4 - Element <AuthzDecisionStatement> | In order for the assertion to be interpreted correctly and securely, the SAML authority and SAML relying party MUST interpret each URI reference in a consistent manner. | Impossible to verify how the SAML authority interprets the URI it provided.
+| 3.2.2 - Complex Type StatusResponseType | If such a signature is used, then the \<ds:Signature> element MUST be present. | Impossible to directly verify since it's impossible to know "If such a signature is used". However the \<ds:Signature> element is tested if provided.
+| 3.3.4 - Processing Rules | If S2 includes one or more \<saml:SubjectConfirmation> elements, then S1 MUST include at least one \<SAML:SubjectConfirmation> element such that S1 can be confirmed in the manner described by at least one \<SAML:SubjectConfirmation> element in S2. | "Confirming" in \<SAML:SubjectConfirmation> is a very difficult task. For now only the \<SAML:SubjectConfirmation> Method attributes are compared, rather than actual confirmation.
+| 3.4.1 - Element <AuthnRequest> | If \[ForceAuthn is\] "true", the identity provider MUST authenticate the presenter directly rather than rely on a previous security context. If a value is not provided, the default is "false". | Hard to test because how they use “previous security context” is implementation-specific. We could catch most cases by handling cookies and/or session variables, for example, but it wouldn’t cover all cases.
```

Comment:
❓ Couldn't we add a test that tries to sign in twice and see if we get the login page twice?